common/src/node/ethereum/provider.rs

```diff
@@ -0,0 +1,74 @@
+use crate::EthError;
+use ethers::providers::{Http, Provider};
+use url::Url;
+
+#[cfg(not(target_arch = "wasm32"))]
+use once_cell::sync::OnceCell;
+#[cfg(not(target_arch = "wasm32"))]
+static G_AGENTINFO: OnceCell<String> = OnceCell::new();
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn set_ethers_httpagent(agent: &str) -> Result<(), EthError> {
+    if G_AGENTINFO.get().is_none() {
+        G_AGENTINFO.set(agent.to_string()).expect("set agent info");
+        return Ok(());
+    }
+    Err(EthError::HttpAgentError)
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn get_ethers_httpagent() -> Result<String, EthError> {
+    if let Some(v) = G_AGENTINFO.get() {
+        return Ok(v.clone());
+    }
+    Err(EthError::HttpAgentError)
+}
```

Comment:
it seems this is just a test code, so can be moved to the test module?

Reply:
ok